### DISCUSSION TITLE
Object with required fields AND additionalProperties == true?

### DISCUSSION THREAD

**Discussion by vlfig** [andyglow/scala-jsonschema#156] [Q&A] [answered]
Hi,

What is the way to define a schema where some properties are required but also other non-specified ones allowed?

I.e. have
```scala
    `object`(
      Field("something", Json.schema[Int])
    ).withAdditionalProperties(true) // does not exist
```
generate something like
```json
{
  "properties" : {
    "something" : { "type" : "integer" }
  },
  "additionalProperties" : true,
  "$schema" : "http://json-schema.org/draft-07/schema#",
  "type" : "object",
  "required" : [
    "something"
  ]
}
```
Cf. https://json-schema.org/understanding-json-schema/reference/object.html#id2

**Comment by vlfig** [comment, marked as answer]
ping @andyglow ?

**Reply by andyglow** [reply]
Hi, @vlfig 
As of now there is no way for partial structure definition
You can either go with strict variant (all fields specified, additionalProperties = false) of free (no fields specified, additionalProperties = true)
There is a chapter in readme on this regard, please check https://github.com/andyglow/scala-jsonschema/#free-objects

Another option would be to list all fields as optional (if this might be the case), or use Free object as nested structure

**Reply by vlfig** [reply]
Thanks. I've created #164 to track this.
